supervisor/api/network.py

```diff
@@ -277,19 +303,25 @@ async def create_vlan(self, request: web.Request) -> None:
         ipv4_setting = None
         if ATTR_IPV4 in body:
             ipv4_setting = IpSetting(
-                body[ATTR_IPV4].get(ATTR_METHOD, InterfaceMethod.AUTO),
-                body[ATTR_IPV4].get(ATTR_ADDRESS, []),
-                body[ATTR_IPV4].get(ATTR_GATEWAY, None),
-                body[ATTR_IPV4].get(ATTR_NAMESERVERS, []),
+                method=body[ATTR_IPV4].get(ATTR_METHOD, InterfaceMethod.AUTO),
+                address=body[ATTR_IPV4].get(ATTR_ADDRESS, []),
+                gateway=body[ATTR_IPV4].get(ATTR_GATEWAY, None),
+                nameservers=body[ATTR_IPV4].get(ATTR_NAMESERVERS, []),
             )
 
         ipv6_setting = None
         if ATTR_IPV6 in body:
-            ipv6_setting = IpSetting(
-                body[ATTR_IPV6].get(ATTR_METHOD, InterfaceMethod.AUTO),
-                body[ATTR_IPV6].get(ATTR_ADDRESS, []),
-                body[ATTR_IPV6].get(ATTR_GATEWAY, None),
-                body[ATTR_IPV6].get(ATTR_NAMESERVERS, []),
+            ipv6_setting = Ip6Setting(
+                method=body[ATTR_IPV6].get(ATTR_METHOD, InterfaceMethod.AUTO),
+                addr_gen_mode=body[ATTR_IPV6].get(
+                    ATTR_ADDR_GEN_MODE, InterfaceAddrGenMode.DEFAULT_OR_EUI64
```

Comment:
and here..

```suggestion
                    ATTR_ADDR_GEN_MODE, InterfaceAddrGenMode.DEFAULT
```

Unless you have good reason to use a deviate from the NetworkManager default for 3 (see https://www.networkmanager.dev/docs/api/1.50.2/settings-ipv6.html) :thinking: .

Reply:
Yeah I though that it could possibly make a mess on an older OS.